lib/streamlit/elements/vega_charts.py

```diff
@@ -519,7 +519,10 @@ def _reset_counter_pattern(prefix: str, vega_spec: str) -> str:
     We need to reset these counters on a spec-level to make the
     spec stable across reruns and avoid changes to the element ID.
     """
-    pattern = re.compile(rf'"{prefix}\d+"')
+
+    # Altair 6.0.0 introduced a new way to handle parameters,
+    # by using hashes instead of pure counters:
+    pattern = re.compile(rf'"{prefix}[0-9a-z]+"')
```

Comment:
Theoretically, altair >=6.0.0 uses stable hashes instead of unstable counters. However, if we would remove this code path for altair >=6.0.0 it would break some apps. Keeping the old logic for now ...